src/main/java/com/inorg/services/product/service/ProductServiceImpl.java

```diff
@@ -131,20 +188,132 @@ public List<Product> createProducts() {
 
     @Override
     public ProductPagedQueryResponse getProductsByCategory(String categoryId) {
-        return null;
-        // apiRoot.products().get()
+       // return null;
+        return  apiRoot.products().get()
+                .addWhere("masterData(current(categories(id = \""+categoryId+"\")))")
                 //TODO withWhere
-                //.executeBlocking().getBody();
+                .executeBlocking().getBody();
     }
 
     @Override
     public ProductProjectionPagedSearchResponse searchProducts(String searchText) {
-        return null;
-       //  apiRoot.productProjections().search()
-                //TODO withText, withFacet
-                //.executeBlocking().getBody();
+        //return null;
+        //TODO withText withFacet
+         return apiRoot.productProjections().search()
+                 .get()
+                 .withText("EN-US", searchText)
+                 .withFacet("variants.attributes.Color:\""+searchText+"\"")
```

Comment:
searchText need to be passed here.

.withFacet("variants.attributes.Color")

Reply:
Result should contain facets like:

![image](https://github.com/sankar-inorg/ServiceProduct/assets/172612410/e021a853-4b31-475e-93d1-caaa56f7aa6c)